core/conn/spj_init/src/main/resources/init_spj.sql

```diff
@@ -0,0 +1,156 @@
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE HELP TO PUBLIC;
+      
+      CREATE PROCEDURE PUT (
+      IN FILEDATA VARCHAR(102400) CHARACTER SET ISO88591,
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      IN CREATEFLAG INTEGER)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.put(java.lang.String,java.lang.String,int)'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE PUT TO PUBLIC;
+      
+      CREATE PROCEDURE LS (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.ls(java.lang.String,java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE LS TO PUBLIC;
+      
+      CREATE PROCEDURE LSALL (
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.lsAll(java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE LSALL TO PUBLIC;
+      
+      CREATE PROCEDURE RM (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.rm(java.lang.String)'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE RM TO PUBLIC;
+      
+      CREATE PROCEDURE RMREX (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.rmRex(java.lang.String, java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE RMREX TO PUBLIC;
+
+      CREATE PROCEDURE GETFILE (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      IN OFFSET INTEGER,
+      OUT FILEDATA VARCHAR(102400) CHARACTER SET ISO88591,
+      OUT DATALENGTH LARGEINT)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.get (java.lang.String,int,java.lang.String[],long[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE GETFILE TO PUBLIC;
+      
+      cqd CAT_IGNORE_ALREADY_EXISTS_ERROR 'off';
+      cqd CAT_IGNORE_DOES_NOT_EXIST_ERROR 'off';
+
+      exit;
+procEOF
+    
+}
+
+dropAndCreateSchema
+createProcedures
+echo "Successfully completed installation of SPJs."
```

Comment:
Good point, I forgot that aspect. The PUT tries to sandbox the files it puts into a specific directory, so it should be relatively safe, but it may still be nice to be able to restrict this to a specific role. Roberta, is it possible that this initialization script creates a role, something like "LIBRARY_CREATOR", so that we can grant execution privilege to that role instead of PUBLIC? Then the DBA of the customer can grant this role to whoever they want.